library/HTMLPurifier/Filter/URIFilter/SafeLink.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+class HTMLPurifier_URIFilter_SafeLink extends HTMLPurifier_URIFilter
+{
+    /**
+     * @type string
+     */
+    public $name = 'SafeLink';
+
+    /**
+     * @type bool
+     */
+    public $always_load = true;
+
+    /**
+     * @type string
+     */
+    protected $regexp = null;
+
+    /**
+     * @param HTMLPurifier_Config $config
+     * @return bool
+     */
+    public function prepare($config)
+    {
+        $this->regexp = $config->get('URI.SafeLinkRegexp');
+
+        return true;
+    }
+
+    /**
+     * @param HTMLPurifier_URI $uri
+     * @param HTMLPurifier_Config $config
+     * @param HTMLPurifier_Context $context
+     * @return bool
+     */
+    public function filter(&$uri, $config, $context)
+    {
+        // check if filter not applicable
+        if ($config->get('HTML.SafeLink') !== true) {
+            return true;
+        }
+
```

Comment:
I don't think the below is needed here because `<link>` is marked as embeds resource: `'href*' => new HTMLPurifier_AttrDef_URI(true),`
```php
if (!$context->get('EmbeddedURI', true)) {
  return true;
}
```

I only removed it because coveralls was failing and I'm not sure how to test it given what I wrote above.